jitbuilder/compile/Method.hpp

```diff
@@ -166,7 +167,7 @@ class ResolvedMethod : public ResolvedMethodBase, public Method
                                TR::SymbolReferenceTable *symRefTab);
 
    protected:
-   const char *_fileName;
+   std::string _fileName;
```

Comment:
Also note that IlBuilder objects other than MethodBuilders and ThunkBuilders should really only be created during compilation (i.e. in the buildIL call of some MethodBuilder). As such, they are actually running as part of the compiler, so I think we should really be trying hard to allocate that memory according to the compiler and not ceding control of those objects to the client.

Granted, MethodBuilder should really be cleaned up on the way out of compileMethodBuilder to eliminate compile-time artifacts like IlBuilder objects (just like I did for SymbolReferences in the TypeDictionary).